MDGameSynchronizer/MDGameSynchronizer.cs

```diff
@@ -297,470 +302,468 @@ private void PauseGame()
-        {
-            MDLog.Debug(LOG_CAT, "All peers are not synched yet");
-            return;
-        }
 
-        bool waitForTickMsec = false;
-
-        // Check if we still need to wait for a better confidence on the TickMsec value
-        foreach (int peerId in GameSession.GetAllPeerIds())
-        {
-            if (peerId == MDStatics.GetServerId())
+            if (!allClientsSynched)
             {
-                continue;
+                MDLog.Debug(LOG_CAT, "All peers are not synched yet");
+                return;
             }
-            if (!InternalTicksList.ContainsKey(peerId) || InternalTicksList[peerId].Count < GetMinimumMeasurementCountBeforeResume())
+
+            bool waitForTickMsec = false;
+
+            // Check if we still need to wait for a better confidence on the TickMsec value
+            foreach (int peerId in GameSession.GetAllPeerIds())
             {
+                if (peerId == MDStatics.GetServerId()
+                    || InternalTicksList.ContainsKey(peerId) &&
+                    InternalTicksList[peerId].Count >= GetMinimumMeasurementCountBeforeResume())
+                {
+                    continue;
+                }
+
                 waitForTickMsec = true;
                 MDLog.Trace(LOG_CAT, "Still waiting for peer [{0}] to get a more secure TickMsec value", peerId);
-                continue;
             }
-        }
 
-        if (waitForTickMsec)
-        {
-            MDLog.Debug(LOG_CAT, "Still waiting for a more secure msec value");
-            return;
-        }
-
-        MDLog.Debug(LOG_CAT, "All clients synched, sending unpause signal");
-        // Alright tell all clients to unpause in a bit
-        foreach (int peerid in GameSession.GetAllPeerIds())
-        {
-            // Get our current game tick
-            uint tickToUnpause = GameClock != null ? GameClock.GetTick() : 0;
-            
-            if (peerid != MDStatics.GetServerId())
+            if (waitForTickMsec)
             {
-                RpcId(peerid, nameof(UnpauseAtTickMsec), GetPlayerTicksMsec(peerid) + GetUnpauseCountdownDurationMSec(), tickToUnpause);
+                MDLog.Debug(LOG_CAT, "Still waiting for a more secure msec value");
+                return;
             }
-        }
 
-        UnpauseAtTickMsec(OS.GetTicksMsec() + GetUnpauseCountdownDurationMSec(), 0);
-        ClientSynchList.Clear();
-        timer.RemoveAndFree();
-    }
-
-    private void OnNetworkNodeAdded(Node node)
-    {
-        MDLog.Trace(LOG_CAT, "Node added: {0}", node.GetPath());
-        NodeList.Add(node);
-    }
+            MDLog.Debug(LOG_CAT, "All clients synched, sending unpause signal");
+            // Alright tell all clients to unpause in a bit
+            foreach (int peerid in GameSession.GetAllPeerIds())
+            {
+                // Get our current game tick
+                uint tickToUnpause = GameClock?.GetTick() ?? 0; // TODO: Is null coalescing is a good thing?
```

Comment:
I'm a fan but my worry is that since it's only available in C# 8 and up some people might not have updated yet.

Reply:
i'm anyway still confused with this a little bit
can't instantly think what will happen with this when looking at it